src/Microsoft.Azure.SignalR.Common/ServiceConnections/MultiEndpointServiceConnectionContainer.cs

```diff
@@ -92,24 +96,24 @@ public Task ConnectionInitializedTask
         {
             get
             {
-                if (_inner != null)
+                if (!_needRouter)
```

Comment:
`_needRouter` need to be in sync with `_inner` otherwise it is not correct. when `_connectionContainers` can change, we need to make sure all the connections added, adding to the container are in control.

Reply:
offline synced to change the way to a little.